dhall/src/Dhall/Import.hs

```diff
@@ -744,8 +746,39 @@ toHeader _ = do
     empty
 
 getCacheFile
-    :: (Alternative m, MonadIO m) => FilePath -> Dhall.Crypto.SHA256Digest -> m FilePath
+    :: (MonadCatch m, Alternative m, MonadIO m) => FilePath -> Dhall.Crypto.SHA256Digest -> m FilePath
 getCacheFile cacheName hash = do
+    cacheDirectory <- getOrCreateCacheDirectory cacheName
+
+    let cacheFile = cacheDirectory </> ("1220" <> show hash)
+
+    return cacheFile
+
+warnAboutMissingCaches :: (MonadCatch m, Alternative m, MonadIO m) => m ()
+warnAboutMissingCaches = warn <|> return ()
+    where warn = Data.Foldable.traverse_ getOrCreateCacheDirectoryOrWarn ["dhall", "dhall-haskell"]
+
+getOrCreateCacheDirectoryOrWarn :: (MonadCatch m, Alternative m, MonadIO m) => FilePath -> m FilePath
+getOrCreateCacheDirectoryOrWarn cacheName = do
+    cacheBaseDir <- getCacheBaseDirectoryOrWarn
+
+    let message =
+            "\n" 
+         <> "\ESC[1;33mWarning\ESC[0m: " 
+         <> "It hasn't been possible to get or create the default cache directory:\n" 
+         <> "\n"
+         <> "  " ++ cacheBaseDir </> cacheName ++ "\n"
+         <> "\n"
+         <> "Usually it is caused by permissions issues.\n"
```

Comment:
I think we can still do more to push the error messages closer to the point of the error.  As a general rule of thumb if we can't confidently diagnose what went wrong then the errors aren't sufficiently localized

Reply:
Mmm agree, i'll give another try, but imho sometimes the added complexity to get more precision for relatively infrequent errors doesn' worth it.
I will inline the warnigs inside the main functions adding a parameter to let `getCacheFile` ignore them.

Reply:
Ok, i've made the inline of warnings within `get*` actions and it will warn about the "expected" errors (directory not having required perms or path pinting to a file instead a directory) and the possible `IOException`'s.
The flag to trigger the warnings is a `Bool` but if we establish a log system like proposed in #1328, it would change to a log level (= WARNING).